### PR TITLE
Script for defect 2462 update logic PTU

### PR DESCRIPTION
**Pull request by TMyza**
ATF Test Scripts to check [#2462](https://github.com/smartdevicelink/sdl_core/issues/2462)

This PR is **ready** for review.

### Summary
A test script to verify that issue is fixed

### ATF version
[develop](https://github.com/smartdevicelink/sdl_atf/tree/develop)

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/Defects/5_0/2462_Need_updated_logic_for_checking_user_consent_prompt_after_PTU.lua [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", common.preconditions)
+runner.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+runner.Step("Register App", common.registerApp)
+runner.Step("Activate App", common.activateApp)
+
+-- [[ Test ]]
+runner.Title("Test")
+runner.Step("RAI, PTU", common.policyTableUpdate, { _, expectationAppPermissionsParam })
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2462_Need_updated_logic_for_checking_user_consent_prompt_after_PTU.lua]: New group with consent was not added.

File: test_scripts/Defects/5_0/2462_Need_updated_logic_for_checking_user_consent_prompt_after_PTU.lua [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+---------------------------------------------------------------------------------------------------
+-- [[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('user_modules/sequences/actions')
+
+-- [[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+-- [[ Local functions ]]
+local function expectationAppPermissionsParam()
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2462_Need_updated_logic_for_checking_user_consent_prompt_after_PTU.lua]: expectation must also contain the expectation `m.getHMIConnection():ExpectNotification("SDL.OnStatusUpdate", { status = "UP_TO_DATE" })`

File: test_scripts/Defects/5_0/2462_Need_updated_logic_for_checking_user_consent_prompt_after_PTU.lua [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+-- 1) Need updated logic for checking user consent prompt after PTU.
+-- Steps to reproduce:
+-- 1) SDL does not check new added group from PTU requires user consent prompt if it was not 
+--    present in local PT.
+-- Actual:
+-- 1) SDL does not send onAppPermissionChange notification with appPermissionsConsentNeeded set to true 
+--    for affected application. As a result permissions popup is shown only after the next application activation.
+-- Expected:
+-- 1) SDL will be check a new added group from PTU requires user consent prompt if it not present in local PT.
+---------------------------------------------------------------------------------------------------
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2462_Need_updated_logic_for_checking_user_consent_prompt_after_PTU.lua]: pay attention that all consents are applicable with EXTERNAL_PROPRIETARY policy

**Comment by dboltovskyi** [comment]
Livio team please do not review this PR for now since it hasn't passed our internal review yet.  
Review and appropriate updates going be done till 1st of November.
Thank you.

**Comment by dboltovskyi** [comment]
Livio team please do not review this PR for now since it hasn't passed our internal review yet.
Review and appropriate updates going be done till 1st of November.
Thank you.

File: test_scripts/Defects/5_0/2462_Need_updated_logic_for_checking_user_consent_prompt_after_PTU.lua [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+-- User story: https://github.com/smartdevicelink/sdl_core/issues/2462
+--
+-- Description:
+-- 1) Need updated logic for checking user consent prompt after PTU.
+-- Steps to reproduce:
+-- 1) SDL does not check new added group from PTU requires user consent prompt if it was not 
+--    present in local PT.
+-- Actual:
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2462_Need_updated_logic_for_checking_user_consent_prompt_after_PTU.lua]: Please remove actual result.

File: test_scripts/Defects/5_0/2462_Need_updated_logic_for_checking_user_consent_prompt_after_PTU.lua [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+-- 1) SDL does not send onAppPermissionChange notification with appPermissionsConsentNeeded set to true 
+--    for affected application. As a result permissions popup is shown only after the next application activation.
+-- Expected:
+-- 1) SDL will be check a new added group from PTU requires user consent prompt if it not present in local PT.
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2462_Need_updated_logic_for_checking_user_consent_prompt_after_PTU.lua]: `SDL will be check` ->  `SDL will check`

**Comment by dboltovskyi** [comment]
Livio team please notice script is ready for review.

**Comment by dboltovskyi** [comment]
@jordynmackool Please notice PR is ready for PM review.

**Review by iCollin — APPROVED** [review]

**Pull request merged**